src/sidebar/config/test/build-settings-test.js

```diff
@@ -126,61 +123,83 @@ describe('sidebar/config/build-settings', () => {
         );
       });
 
-      it('creates merged settings when the RPC requests returns the host config', async () => {
-        const appConfig = { foo: 'bar', appType: 'via' };
+      it('adds RPC settings to resulting SidebarSettings', async () => {
         fakeJsonRpc.call.resolves({ foo: 'baz' }); // host config
-        const result = await buildSettings(appConfig, fakeWindow);
+
+        const result = await buildSettings({}, fakeWindow);
+
+        assert.deepEqual(result.rpc, {
+          origin: 'https://embedder.com',
+          targetFrame: fakeTopWindow,
+        });
+      });
+
+      it('merges ConfigFromHost returned from RPC with ConfigFromSidebar', async () => {
+        const configFromSidebar = { foo: 'bar', appType: 'via' };
+        fakeJsonRpc.call.resolves({ foo: 'baz' }); // host config
+        const result = await buildSettings(configFromSidebar, fakeWindow);
         assert.deepEqual(result, {
           foo: 'baz',
           appType: 'via',
           apiUrl: fakeApiUrl(),
+          rpc: {
+            origin: 'https://embedder.com',
+            targetFrame: fakeTopWindow,
+          },
         });
       });
 
-      it('rejects if fetching host config fails` ', async () => {
+      it('rejects if RPC request for ConfigFromHost fails', async () => {
         fakeJsonRpc.call.rejects(new Error('Nope'));
-        const appConfig = { foo: 'bar', appType: 'via' };
-        await assert.rejects(buildSettings(appConfig, fakeWindow), 'Nope');
+        const configFromSidebar = { foo: 'bar', appType: 'via' };
+        await assert.rejects(
+          buildSettings(configFromSidebar, fakeWindow),
+          'Nope'
+        );
       });
 
       it('returns the `groups` array with the initial host config request', async () => {
-        const appConfig = {
+        const configFromSidebar = {
           services: [{ groups: ['group1', 'group2'] }],
           appType: 'via',
         };
         fakeJsonRpc.call.onFirstCall().resolves({ foo: 'baz' }); // host config
-        const result = await buildSettings(appConfig, fakeWindow);
+        const result = await buildSettings(configFromSidebar, fakeWindow);
         assert.deepEqual(result.services[0].groups, ['group1', 'group2']);
       });
 
       it("creates merged settings where `groups` is a promise when its initial value is '$rpc:requestGroups'", async () => {
-        const appConfig = {
-          services: [{ groups: '$rpc:requestGroups' }],
+        const configFromSidebar = {
```

Comment:
The updated implementation of `buildSettings` will not accept (or, that is, act upon and fetch) `$rpc:requestGroups` values that are not from _remotely-fetched ConfigFromHost objects_. i.e. they will be ignored on `ConfigFromSidebar` objects. This is a change from the previous behavior but I think it is correct.